File: Client/flavor-haven/src/app/models/dtos/order/order.dto.ts
import { OrderStatusDTO } from './order-status.dto';
import { OrderItemDTO } from './order-item.dto';

export interface OrderDTO {
  id: string;
  orderNumber: string;
  createdTime: Date;
  amount: number;
  note: string;
  address: string;
  isPayed: boolean;
  userId:string;
  status: OrderStatusDTO;
  orderItems: OrderItemDTO[];
}

File: Backend/FlavorHaven/FlavorHaven.Application/Models/DTOs/OrderDTO.cs
namespace FlavorHaven.Application.Models.DTOs;

public class OrderDTO
{
    public Guid Id { get; set; }
    public string OrderNumber { get; set; }
    public DateTime CreatedTime { get; set; }
    public decimal Amount { get; set; }
    public string Note { get; set; }
    public string Address { get; set; }
    public bool IsPayed { get; set; }
    public Guid UserId { get; set; }
    
    public OrderStatusDTO Status { get; set; }
    
    public ICollection<OrderItemDTO> OrderItems { get; set; }
}